### DISCUSSION TITLE
Becoming Stable

### DISCUSSION THREAD

**Discussion by mara-schulke** [helsing-ai/buffrs#122] [Roadmaps]
This is a open discussion on the features (to be implemented), issues / bugs (that have to be fixed), decisions to be made etc. until we can consider Buffrs to be stable enough for public usage, and longterm support cycles.

Currently as we are pre-1.0 – breaking changes still occur and are done so on purpose to reduce technical debt. Soon enough we will be unable to do this, so writing up a list of breaking changes we need to make now is the first step to improve stability and foster the usage of Buffrs as a platform.

---

### Features

- [ ] Support OCI registries to become selfhotable
  - #229
- [ ] Dynamic Version Resolution Using Semver + Pubgrub
  - #205
  - #234
  - #235
- [ ] Cross-Registry Dependencies / Distributed Package Management
- [ ] Git Dependencies
  - #73
- [ ] Local Filesystem Dependencies
- [ ] SSO Registry Authentication
  - #16
- [x] Introducing Manifest and Lockfile editions

### Housekeeping

- [ ] Issue Templates
- [ ] Contributing Notice
- [ ] PM Setup in GitHub
- [ ] Really Good Documentation and Learning Resources

### Integrations

- [x] Nix
- [ ] Rust / Cargo
  - #54
  - #43 
  - #55 
- [ ] Python / Poetry 
  - #27
- [ ] JS / TS

### Security

- [ ] Credential Storage 
  - #109
  - #119

### Datamodel

- [ ] Standardizing The Buffrs Package Format

**Comment by qsantos** [comment]
Regarding the first point, I have taken the liberty of opening https://github.com/helsing-ai/buffrs/issues/205.
